internal/auth0/user/resource_permission.go

```diff
@@ -0,0 +1,169 @@
+				Computed:    true,
+				Description: "Description of the permission.",
+			},
+			"resource_server_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the resource server that the permission is associated with.",
+			},
+		},
+		CreateContext: createUserPermission,
+		ReadContext:   readUserPermission,
+		DeleteContext: deleteUserPermission,
+		Importer: &schema.ResourceImporter{
+			StateContext: importUserPermission,
+		},
+		Description: "With this resource, you can manage user permissions.",
+	}
+}
+
+func createUserPermission(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*management.Management)
+
+	userId := data.Get("user_id").(string)
+	resourceServerId := data.Get("resource_server_identifier").(string)
+	permissionName := data.Get("permission").(string)
+
+	mutex.Global.Lock(userId)
+	defer mutex.Global.Unlock(userId)
+
+	if err := api.User.AssignPermissions(userId, []*management.Permission{
+		{
+			ResourceServerIdentifier: &resourceServerId,
+			Name:                     &permissionName,
+		},
+	}); err != nil {
+		return diag.FromErr(err)
+	}
+
+	data.SetId(userId + resourceServerId + permissionName)
+
+	return readUserPermission(ctx, data, meta)
+}
+
+func readUserPermission(_ context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*management.Management)
+
+	userId := data.Get("user_id").(string)
+	permissionName := data.Get("permission").(string)
+	resourceServerId := data.Get("resource_server_identifier").(string)
+
+	existingPermissions, err := api.User.Permissions(userId)
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	for _, p := range existingPermissions.Permissions {
+		if p.GetName() == permissionName && p.GetResourceServerIdentifier() == resourceServerId {
+			data.Set("description", p.GetDescription())
+			data.Set("resource_server_name", p.GetResourceServerName())
+			return nil
+		}
+	}
+
+	data.SetId("")
+	return nil
+
+}
+
+func deleteUserPermission(_ context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*management.Management)
+
+	userId := data.Get("user_id").(string)
+	permissionName := data.Get("permission").(string)
+	resourceServerId := data.Get("resource_server_identifier").(string)
+
+	mutex.Global.Lock(userId)
+	defer mutex.Global.Unlock(userId)
+
+	if err := api.User.RemovePermissions(
+		userId,
+		[]*management.Permission{
+			{
+				ResourceServerIdentifier: &resourceServerId,
+				Name:                     &permissionName,
+			},
+		},
+	); err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	data.SetId("")
+	return nil
+}
+
+func importUserPermission(
```

Comment:
Happy to go forward with this, but it would be nice if we have a follow up PR where we change the schema.ImportResourceIDPair func to accept a variadic list of resource attribute names.